taskcluster/ac_taskgraph/release_promotion.py

```diff
@@ -0,0 +1,172 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import os
+
+from mozilla_version.fenix import FenixVersion
+from taskgraph.actions.registry import register_callback_action
+
+from taskgraph.util.taskcluster import get_artifact
+from taskgraph.taskgraph import TaskGraph
+from taskgraph.decision import taskgraph_decision
+from taskgraph.parameters import Parameters
+from taskgraph.util.taskgraph import find_decision_task, find_existing_tasks_from_previous_kinds
+
+RELEASE_PROMOTION_PROJECTS = (
+    "https://github.com/mozilla-mobile/android-components",
+)
+
+
+def is_release_promotion_available(parameters):
+    return parameters['head_repository'] in RELEASE_PROMOTION_PROJECTS
+
+
+@register_callback_action(
+    name='release-promotion',
+    title='Ship Android Components',
+    symbol='${input.release_promotion_flavor}',
+    description="Ship Android Components",
+    generic=False,
+    order=500,
+    context=[],
+    available=is_release_promotion_available,
+    schema=lambda graph_config: {
+        'type': 'object',
+        'properties': {
+            'build_number': {
+                'type': 'integer',
+                'default': 1,
+                'minimum': 1,
+                'title': 'The release build number',
+                'description': ('The release build number. Starts at 1 per '
+                                'release version, and increments on rebuild.'),
+            },
+            'do_not_optimize': {
+                'type': 'array',
+                'description': ('Optional: a list of labels to avoid optimizing out '
+                                'of the graph (to force a rerun of, say, '
+                                'funsize docker-image tasks).'),
+                'items': {
+                    'type': 'string',
+                },
+            },
+            'revision': {
+                'type': 'string',
+                'title': 'Optional: revision to ship',
+                'description': ('Optional: the revision to ship.'),
+            },
+            'release_promotion_flavor': {
+                'type': 'string',
+                'description': 'The flavor of release promotion to perform.',
+                'default': 'build',
+                'enum': sorted(graph_config['release-promotion']['flavors'].keys()),
```

Comment:
Ah. To define `graph_config['release-promotion']['flavors']`, we probably need to copy [this block](https://github.com/mozilla-mobile/fenix/blob/master/taskcluster/ci/config.yml#L96-L99) into `taskcluster/ci/config.yml` in android-components.

Reply:
Thanks for the pointer!